Feat: Appreg v2 updates (#6855)
Kong/docs.konghq.com: app/konnect/dev-portal/applications/enable-app-reg.md

Reviewer: Does it make sense to add API instructions here?

When you apply an auth strategy, {{site.konnect_saas}} enables plugins automatically to support the desired mode, either key authentication or OpenID Connect.
These plugins run inside the data plane to support application registration for the API product version and are managed by
{{site.konnect_saas}}.

Enabling application registration is specific to the [geographic region](/konnect/geo). 
You must enable application registration in each geo that you want to allow developers to register with.
Each geo has their own API keys and specifications for application registration in their respective geo.

{{site.konnect_short_name}} provides the capability to configure and implement a range of authentication strategies. Utilizing the OpenID Connect authentication strategy allows for specific permissions to be set within your Identity Provider (IdP). This system offers the versatility to either apply a unified authentication strategy across all API Products or to designate a unique authentication strategy to individual API Products.

> 

Reviewer: ```suggestion
<!--vale off-->
```

{% mermaid %}
flowchart TB
    subgraph Staging Portal
    WeatherAPIv1["Weather API v1"] --> key-auth
    WeatherAPIv2["Weather API v2"] --> okta-oidc
    MapsAPIv2-1["Maps API v2"] --> okta-oidc
    end
    subgraph Production Portal
    WeatherAPIv3["Weather API v3"] --> okta-dcr
    MapsAPIv2-2["Maps API v2"] --> auth0-oidc
    end

{% endmermaid %}

> 

Reviewer: ```suggestion
<!--vale on-->
```

content_type: how-to
---

> To grant developers access to [register an application](/konnect/dev-portal/applications/dev-reg-app-service/), you must enable application registration for an API product version. 
> When you enable application registration, {{site.konnect_saas}} enables plugins automatically to support the desired mode, either key authentication or OpenID Connect.

Reviewer 2: ```suggestion
To grant developers access to [register an application](/konnect/dev-portal/applications/dev-reg-app-service/), you must apply an authentication strategy and enable application registration for an API product version. 
When you apply an auth strategy, {{site.konnect_saas}} enables plugins automatically to support the desired mode, either key authentication or OpenID Connect.
```

You must enable application registration in each geo that you want to allow developers to register with.
Each geo has their own API keys and specifications for application registration in their respective geo.

> The authentication strategies that facilitate application registration correspond to specific permissions within your Identity Provider (IdP). {{site.konnect_short_name}} enables you to develop and apply various authentication strategies. You have the flexibility to either implement a single auth strategy across all your API Products or to assign a distinct auth strategy for each API Product.

Reviewer 2: ```suggestion
{{site.konnect_short_name}} enables you to configure and apply various auth strategies. When using an OpenID Connect auth strategy, specific permissions can be configured within your Identity Provider (IdP). You have the flexibility to either implement a single auth strategy across all your API Products or to assign a distinct auth strategy for each API Product.
```

   | `Auto Approve`| **Default: disabled** <br>Automatically approve developer application requests for an application.| **False**

   
## Enable app registration with multiple IdPs

In {{site.konnect_short_name}} can configure and manage multiple authentication strategies across various API products and their versions, allowing you to apply distinct authentication scopes for different API versions.

> This section will introduce you to the functionality portal product versions using Dynamic client registration (DCR). Using the Application Registration API, you can manage multiple APIs and configure different DCRs on a per API product basis.

Reviewer: ```suggestion
This section will introduce you to the functionality portal product versions using Dynamic client registration (DCR). Using the Application Registration API, you can manage multiple APIs and configure a different DCR on a per API product basis.
```

1. Configure the auth strategies of your choice:
  * [Okta](/konnect/dev-portal/applications/dynamic-client-registration/okta/)
  * [Curity](/konnect/dev-portal/applications/dynamic-client-registration/curity/)
  * [Auth0](/konnect/dev-portal/applications/dynamic-client-registration/auth0/)
  * [Azure](/konnect/dev-portal/applications/dynamic-client-registration/azure/)
  * [Custom IdP](/konnect/dev-portal/applications/dynamic-client-registration/custom/)

1. Before making a request to this endpoint, ensure you have gathered the following details from the previous step:
  * `PortalId`: ID of the Dev Portal
  * `product_version_ids`: API Product version ID.
  * `auth_strategy_id`: The ID of the auth strategies you configured.

1. Apply the auth strategy to your API, making sure to replace `your_product_version_id` and `your_auth_strategy_id` with the actual IDs you intend to use: 

  ```bash
  curl --request POST \

>     --url https://us.api.konghq.tech/v2/portals/{portalId}/product-versions/ \

Author: ```suggestion
    --url https://us.api.konghq.com/v2/portals/{portalId}/product-versions/ \
```
